mtags/src/main/scala-3/scala/meta/internal/pc/printer/DotcPrinter.scala

```diff
@@ -41,6 +42,22 @@ object DotcPrinter:
 
     def fullName(sym: Symbol): String =
       fullNameString(sym)
+
+    override def toText(tp: Type): Text =
+      // Override the behavior for `AppliedType` because
+      // `toText` in RefinedPrinter doesn't pretty print AppliedType
+      // if tycon is instance of PrettyType.
+      // For example, if we don't override `toText`,
+      // CompletionoverrideConfigSuite's `package` test will fail with
+      // completing `def function: Int <none> String = ${0:???}`
+      // instead of `def function: f.Function[Int, String] = ${0:???}`
+      tp match
+        case tp: AppliedType =>
+          tp.tycon match
+            case p: PrettyType =>
```

Comment:
Overriding it only for PrettyType wouldn't work?

Reply:
No, if we have `AppliedType(PrettyType("Foo"), Seq(Int, String))`, `DocPrinter` will handle `AppliedType` using [RefinedPrinter's toText](https://github.com/lampepfl/dotty/blob/9e3953d53c815a61f85311e4863f0431690063d8/compiler/src/dotty/tools/dotc/printing/RefinedPrinter.scala#L140-L300) and it never call this method for `PrettyType` recursively, we will end up having `Int <none> String` instead of `Foo[Int, String]`.